packages/common/src/codegen/render-solidity/renderTypeHelpers.ts

```diff
@@ -63,9 +63,15 @@ function renderWrapperStaticArray(
     ) pure returns (
       ${elementType}[${staticLength}] memory _result
     ) {
-      // in memory static arrays are just dynamic arrays without the length byte
-      assembly {
-        _result := add(_value, 0x20)
+      if (_value.length < ${staticLength}) {
+        // ignore invalid dynamic arrays that are too small
+        return _result;
+      } else {
+        // in memory static arrays are just dynamic arrays without the length byte
+        // (without the length check this could lead to memory corruption)
+        assembly {
+          _result := add(_value, 0x20)
+        }
```

Comment:
any way we can add a regression test for this to make sure we don't run into this in the future?

Reply:
memory corruption tests are tricky, I think I'm too sleepy to write 1 rn, we can add an issue for later, tests aren't production code so does freeze affect em?

Reply:
we can always add more tests later! planning to do extend the test suite in general next week

Reply:
added here: https://github.com/latticexyz/mud/issues/1603